mantidproject/mantid
# Fix performance regression in ConvexPolygonTestPerformance

Description of work.

This addresses the `ConvexPolygonlTestPerformance` and `QuadrilateralTestPerformance` regressions discussed in #15856. with GCC on linux, switching Vertices from a `std::deque` to a `std::vector` eliminates calls to `malloc_consolidate` and improves performance in the old test by about an 10x. 

I split up the performance tests into two, one that updates the points in a `ConvexPolygon` and another that tests calculating the area of the polyhedron. 

**To test:**



This PR is best verified on Linux since the performance regression occurs there. Since I modified the performance tests, use the test currently in master to verify that the performance regression is fixed. 

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
